### PR TITLE
DM-5684: Unused variables in meas_extensions_psfex library code

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: python/lsst/meas/extensions/psfex/psfex.py [hunk 1]
```diff
@@ -651,7 +283,7 @@ def showPsf(psf, set, ext=None, wcsData=None, trim=0, nspot=5,
         mosaic.writeFits(outFile)
 
 
-def getLsstFlags(tab=None):
+def getFlags(tab=None):
     flagKeys = [
         "base_PixelFlags_flag_edge",
         # "base_PixelFlags_flag_interpolated",
```
Comment (kgabor) [inline python/lsst/meas/extensions/psfex/psfex.py:289]: Do we want to keep these commented out flags?

File: python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py [hunk 2]
```diff
@@ -236,14 +236,9 @@ def determinePsf(self, exposure, psfCandidateList, metadata=None, flagKey=None):
             gain = 1.0
             self.log.warn("Setting gain to %g" % (gain,))
```
Comment (kgabor) [inline python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py:237]: For consistency log.warn("Setting gain to %g", gain) ?

**Review by kgabor — APPROVED** [review]
Additional minor comments at not directly affected locations:
`psfexPsfDeterminer.py`:193 : Do we want to keep a print() call here even in case of "display" mode?
`psfexPsfDeterminer.py`:245, 252: "* Error *" emphasis is unnecessary in an exception. Also, do we want to replace KeyError to RuntimeError?

File: python/lsst/meas/extensions/psfex/psfexStarSelector.py [hunk 3]
```diff
@@ -70,11 +70,13 @@ class PsfexStarSelectorConfig(BaseSourceSelectorTask.ConfigClass):
         doc="Max number of bad pixels ",
         default=0,
         check=lambda x: x >= 0,
+        deprecated="This field has never worked. Will be removed after v21."
```
Comment (kgabor) [inline python/lsst/meas/extensions/psfex/psfexStarSelector.py:73]: In the deprecation message state that this config option is actually not used anywhere in the code any more.

File: python/lsst/meas/extensions/psfex/psfex.py (hunk 1 above)
Comment (parejkoj) [reply python/lsst/meas/extensions/psfex/psfex.py:289]: Hmm... I think I'll leave them for future reference. There's no useful info in the git blame history, so I don't know why they were commented out.

File: python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py (hunk 2 above)
Comment (parejkoj) [reply python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py:237]: Haha! Good catch, but I'm leaving it. I think the intention here is to warn if it couldn't get a detector. Whether that ever happens any more, I truly doubt, but I'm not going to change it here.

File: python/lsst/meas/extensions/psfex/psfexStarSelector.py (hunk 3 above)
Comment (parejkoj) [reply python/lsst/meas/extensions/psfex/psfexStarSelector.py:73]: I mean, if anyone had set it, they'd have gotten an `AssertionError`, but I'll expand the message.

**Comment by parejkoj** [comment]
Thanks for the other general comments: I incorporated those into new commits.

**Pull request merged**